proc/threads.c

```diff
@@ -1591,6 +1612,10 @@ static int _proc_lockSet(lock_t *lock, int interruptible, spinlock_ctx_t *scp)
 		hal_spinlockClear(&threads_common.spinlock, &sc);
 	}
 
+	if (lock->attr.type == LOCK_TYPE_RECURSIVE) {
+		lock->depth = 1;
+	}
```

Comment:
`If` is somewhat redundant 
```suggestion
	lock->depth = 1;
```